torch/_tensor.py

```diff
@@ -223,7 +223,11 @@ def _reduce_ex_internal(self, proto):
         #    `tolist()` converts every single element in the tensor into python objects
         #    and serialize them one by one.
         if self.device.type in ['xla', 'ort', 'mps', 'hpu']:
-            return (torch._utils._rebuild_device_tensor_from_numpy, (self.cpu().numpy(),
+            # Convert BFloat16 tesors to Float32 before conversion to numpy, as numpy doesn't
```

Comment:
Can we add a test to make sure this works as expected?

Reply:
Hi @albanD , I have a test case in test_serialization.py but was not sure which device to use it for. CPU/CUDA serialization goes through a different path. I can try with xla, but I have a question. Can I add xla as a device to the test_serialization.py as is or do I need to co-ordinate with torch_xla?

Reply:
cc @JackCaoG is that something that we could test on the xla CI to make sure we don't regress?

Reply:
Thanks @albanD, I can add a test on xla end.

Reply:
Thanks @JackCaoG

Reply:
@albanD - let me know if there is anything else required or if we can proceed with merging this patch.